src/flb_network.c

```diff
@@ -186,6 +190,42 @@ int flb_net_socket_tcp_fastopen(flb_sockfd_t fd)
     return setsockopt(fd, SOL_TCP, TCP_FASTOPEN, &qlen, sizeof(qlen));
 }
 
+
+/*
+ * Enable TCP keepalive
+ */
+int flb_net_socket_tcp_keepalive(flb_sockfd_t fd, struct flb_net_setup *net)
```

Comment:
Please replace the bitwise assignments of `ret` with regular assignments and add `ret == 0 &&` as a prefix to the conditional blocks so a failed `setsockopt` call is not overwritten and thus missed by the conditional in line 221.

Please change the conditional in line 221 so it explicitly compares against zero (ie. `if (ret != 0) {`).

Reply:
This is important because apparently according to msdn, windows doesn't accept values larger than `255` for `TCP_KEEPCNT` whereas other operating systems accept much larger values,.